PerfettoCds/Pipeline/PerfettoSourceParser.cs

```diff
@@ -0,0 +1,216 @@
+                                throw new Exception("Invalid event type");
+                        }
+                    }
+
+                    var colIndex = cellCount % numColumns;
+                    var colName = cols[colIndex].ToLower();
+
+                    // The event itself is responsible for figuring out how to process and store cell contents
+                    ev.ProcessCell(colName, cell, batch, stringCells, cellCounters);
+
+                    // If we've reached the end of a row, we've finished an event. Store it.
+                    if (++cellCount % numColumns == 0)
+                    {
+                        if (ev.GetType() == typeof(PerfettoSliceEvent))
+                        {
+                            // We get the timestamps used for displaying these events from the slice event
+                            if (traceStartTime == null)
+                            {
+                                traceStartTime = ((PerfettoSliceEvent)ev).Timestamp;
+                            }
+                            traceEndTime = ((PerfettoSliceEvent)ev).Timestamp;
+                        }
+
+                        // Store the event
+                        var result = dataProcessor.ProcessDataElement(ev, this, cancellationToken);
+
+                        ev = null;
+                    }
+                    IncreaseProgress(progressIncrease);
+                }
+            }
+
+            if (traceStartTime.HasValue)
+            {
+                // Use DateTime.Now as the wall clock time. This doesn't matter for displaying events on a relative timescale
+                // TODO Actual wall clock time needs to be gathered from SQL somehow
+                this.dataSourceInfo = new DataSourceInfo(traceStartTime.Value.ToNanoseconds, traceEndTime.Value.ToNanoseconds, DateTime.Now.ToUniversalTime());
+            }
+        }
+
+
+        public void ProcessSource(ISourceDataProcessor<PerfettoSqlEvent, PerfettoSourceParser, string> dataProcessor,
+            ILogger logger,
+            IProgress<int> progress,
+            CancellationToken cancellationToken)
+        {
+            this.Progress = progress;
+
+            PerfettoTraceProcessor traceProc = new PerfettoTraceProcessor();
+
+            try
+            {
+                // Start the progress counter to indicate something is happening because
+                // OpenTraceProcessor could take a few seconds
+                IncreaseProgress(1);
+
+                // Start the Perfetto trace processor shell with the trace file
+                traceProc.OpenTraceProcessor(filePath);
+
+                // We're saying the SQL queries take up about 50 percent of the processing
+                double queryProgressIncrease = 49.0 / 5.0; // We're doing 5 SQL queries below
+                
+                // Run queries over all the tables we care about
+                var sliceQr = traceProc.QueryTrace(PerfettoSliceEvent.SqlQuery);
```

Comment:
maybe add logging statements here for the queries we are doing

Reply:
Using this logger?            

` ILogger logger
`